findbugs/src/java/edu/umd/cs/findbugs/detect/FindRefComparison.java

```diff
@@ -967,16 +969,32 @@ private void checkRefComparison(Location location, JavaClass jclass, Method meth
             if (lhsType.equals(Type.OBJECT) && rhsType.equals(Type.OBJECT)) {
                 return;
             }
-            String lhs = SignatureConverter.convert(lhsType.getSignature());
-            String rhs = SignatureConverter.convert(rhsType.getSignature());
 
-            if ("java.lang.String".equals(lhs) || "java.lang.String".equals(rhs)) {
+            String lhsSignature = lhsType.getSignature();
+            @SlashedClassName
+            String lhsSlashed = ClassName.extractClassName(lhsSignature);
+            @DottedClassName
+            String lhsDotted = SignatureConverter.convert(lhsSignature);
+
+            String rhsSignature = rhsType.getSignature();
+            @SlashedClassName
+            String rhsSlashed = ClassName.extractClassName(rhsSignature);
```

Comment:
Currently your code reports missing classes like "C", "Z", "I". I suspect that's because extractClassName leaves the argument as is if the class is primitive type and you try to load it in "isAnnotatedAsValueBasedClass". Try to use ClassName.fromFieldSignature(..) (you will probably need to check the result for null). Running ant in findbugsTestCases should not report any errors, nor any missing classes.